src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -939,7 +939,7 @@ void ControlBoardWrapper::run()
 //
 bool ControlBoardWrapper::setPid(int j, const Pid &p)
 {
-    int off=device.lut[j].offset;
+    int off; try{off = device.lut.at(j).offset;} catch(...){yError() << "joint number " << j <<  " out of bound [0-"<< controlledJoints << "] for part " << partName; return false; }
```

Comment:
If defining `better` we include also readability of the code, and I think that some good old plain `if` would be much more easier to read. 

If instead we are only interested in time complexity, I am not sure at all that the `.at()` based implementation is always better w.r.t. to time complexity. 

An example in which it could be much worse is if `.at()` is not be proper inlined by the compiler (and modern compilers are not magically always taking the right decision with respect to inlining, despite popular belief). In that case, just getting the code for `.at()` could create an `L1` cache miss, that for sure would be much more costly that a simple `if` comparison between integers present in the register or the cache. 

I am not saying that this would happen for sure, but just claiming that we can understand which one of the two constucts is faster on modern systems is totally unrealistic (and probably performance is not a big problem anyway in this case).